File: src/main/java/com/kelsos/mbrc/commands/visual/UpdatePlaybackPositionCommand.java
package com.kelsos.mbrc.commands.visual;

import android.app.Activity;
import com.google.inject.Inject;
import com.kelsos.mbrc.controller.ActiveFragmentProvider;
import com.kelsos.mbrc.fragments.MainFragment;
import com.kelsos.mbrc.interfaces.ICommand;
import com.kelsos.mbrc.interfaces.IEvent;
import org.codehaus.jackson.node.ObjectNode;

import java.util.LinkedHashMap;

public class UpdatePlaybackPositionCommand implements ICommand
{
	@Inject
	ActiveFragmentProvider afProvider;

	public void execute(IEvent e)
	{

        ObjectNode oNode = (ObjectNode)e.getData();

		final int current = oNode.path("current").asInt();
		final int total = oNode.path("total").asInt();
		if (afProvider.getActiveFragment(MainFragment.class) != null)
		{
			Activity cActivity = afProvider.getActiveFragment(MainFragment.class).getActivity();
			cActivity.runOnUiThread(new Runnable()
			{
				@Override
				public void run()
				{
					((MainFragment) afProvider.getActiveFragment(MainFragment.class)).updateDurationDisplay(current, total);;
				}
			});
		}
	}
}

File: app/src/main/kotlin/com/kelsos/mbrc/commands/visual/UpdatePlaybackPositionCommand.kt
package com.kelsos.mbrc.commands.visual

import com.fasterxml.jackson.databind.node.ObjectNode
import com.kelsos.mbrc.events.bus.RxBus
import com.kelsos.mbrc.events.ui.UpdatePosition
import com.kelsos.mbrc.interfaces.ICommand
import com.kelsos.mbrc.interfaces.IEvent
import javax.inject.Inject

class UpdatePlaybackPositionCommand
@Inject constructor(private val bus: RxBus) : ICommand {

  override fun execute(e: IEvent) {
    val oNode = e.data as ObjectNode
    bus.post(UpdatePosition(oNode.path("current").asInt(), oNode.path("total").asInt()))
  }
}
